backend/src/agents/agent.py

```diff
@@ -1,41 +1,58 @@
 from abc import ABC
+import json
 import logging
 from typing import List, Type
-from src.utils import call_model_with_tools
-from .adapters import convert_to_mistral_tool
+
+from .adapters import create_all_tools_str, extract_tool, validate_args
+from src.utils import call_model, get_scratchpad
+from src.prompts import PromptEngine
 from .tool import Tool
 from .types import Action_and_args
 
+engine = PromptEngine()
 
 class Agent(ABC):
     name: str
     description: str
     tools: List[Tool]
-    prompt: str
-
     def __get_action(self, utterance: str) -> Action_and_args:
-        tools = map(convert_to_mistral_tool, self.tools)
-        (function_name, function_params) = call_model_with_tools(self.prompt, utterance, tools)
 
-        tool = next((tool for tool in self.tools if tool.action.__name__ == function_name), None)
+        format_prompt = engine.load_prompt("tool-selection-format")

```

Comment:
could live outside of this class, right below the engine initialisation on line 12, to avoid being re created on every __get_action call

Reply:
Yes, agreed.

Moved outside the class